contracts/ERC1155NFT.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract ERC1155NFT is ERC1155, Ownable {
+
+    mapping(uint256 => uint256) private balances;
+
+    mapping(uint256 => address[]) private owners;
+
+    constructor(string memory _uri) ERC1155(_uri) {}
+
+    function setURI(string memory newuri) public onlyOwner {
+        _setURI(newuri);
+    }
+
+    function mint(
+        address account,
+        uint256 id,
+        uint256 amount,
+        bytes memory data
```

Comment:
Would it actually make sense to have an optional(?) tokenUri param here as well? Having a list of tokenuris on the contract level and overriding the IERC1155MetadataURI uri() function? (Again example from opensea contract)

Reply:
Yah.. its good way to setup tokenUri while token mint from users side, So just added, Also done with tokenuris on contract level.

Reply:
Solved